jetty-core/jetty-websocket/jetty-websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/WebSocketServerComponents.java

```diff
@@ -126,7 +158,16 @@ public String toString()
 
     public static WebSocketComponents getWebSocketComponents(ContextHandler contextHandler)
     {
-        WebSocketComponents components = (WebSocketComponents)contextHandler.getAttribute(WEBSOCKET_COMPONENTS_ATTRIBUTE);
+        WebSocketComponents components = (WebSocketComponents)contextHandler.getContext().getAttribute(WEBSOCKET_COMPONENTS_ATTRIBUTE);
+        if (components == null)
+            throw new IllegalStateException("WebSocketComponents has not been created");
+
+        return components;
+    }
+
+    public static WebSocketComponents getWebSocketComponents(Server server)
+    {
+        WebSocketComponents components = (WebSocketComponents)server.getAttribute(WEBSOCKET_COMPONENTS_ATTRIBUTE);
```

Comment:
Refactor both methods to call a private method that takes `Attributes` as parameter?